tools/LibTargets.mk

```diff
@@ -188,10 +188,10 @@ $(ARCH_SRC)$(DELIM)libuarch$(LIBEXT): context
 staging$(DELIM)libuarch$(LIBEXT): $(ARCH_SRC)$(DELIM)libuarch$(LIBEXT)
 	$(Q) $(call INSTALL_LIB,$<,$@)
 
-libs$(DELIM)libxx$(DELIM)$(LIBXX)$(LIBEXT): context
-	$(Q) $(MAKE) -C libs$(DELIM)libxx TOPDIR="$(TOPDIR)" $(LIBXX)$(LIBEXT) KERNEL=n
+libs$(DELIM)libxx$(DELIM)libxx$(LIBEXT): context
+	$(Q) $(MAKE) -C libs$(DELIM)libxx TOPDIR="$(TOPDIR)" libxx$(LIBEXT) KERNEL=n
 
-staging$(DELIM)$(LIBXX)$(LIBEXT): libs$(DELIM)libxx$(DELIM)$(LIBXX)$(LIBEXT)
+staging$(DELIM)libxx$(LIBEXT): libs$(DELIM)libxx$(DELIM)libxx$(LIBEXT)
 	$(Q) $(call INSTALL_LIB,$<,$@)
```

Comment:
Are you sure this is not correct?  If you are using uClibc++, it installs one location.  if you are using libcxx, it installs at a different location.  Have you tried this with both C++ libraries to assure that it does not break the build.  I don't think we can merge this if we do not have that confidence.

Reply:
Yes, the different C++ library install into the different folder. Make.defs in each folder can only control which C++ source files need to compile, but the final achieved library is controlled by libs/libxx/Makefile:
ifeq ($(CONFIG_UCLIBCXX),y)
include uClibc++/Make.defs
endif

ifeq ($(CONFIG_LIBCXX),y)
include libcxx/Make.defs
endif

# Object Files

AOBJS = $(ASRCS:.S=$(OBJEXT))
COBJS = $(CSRCS:.c=$(OBJEXT))
CXXOBJS = $(CXXSRCS:.cxx=$(OBJEXT))

SRCS = $(ASRCS) $(CSRCS) $(CXXSRCS)
OBJS = $(AOBJS) $(COBJS) $(CXXOBJS)

BIN = libxx$(LIBEXT)

all: $(BIN)

So the final library is determinated by BIN variable.  It's more simple to make BIN equal libxx.
I test this patch with two configuration:
1.sim with uClibc++
2.m4 with libcxx
Both build without error.

Reply:
Okay.  Thanks.  I just wanted be be sure that you took that into consideration.  I will merge this one too.